src/bullsandcows/Main.java

```diff
@@ -1,7 +1,194 @@
+        if (code < 10)
+        {
+            return String.valueOf(code).charAt(0);
+        } else
+        {
+            return (char) (code - 9 + 'a');
+        }
+    }
+
+    public static String borderDigits(int code) {
+        if (code < 2)
+        {
+            return code + "";
+        }
+
+        return 0 + "-" + Math.min(9, code-1);
+    }
+
+    public static String borderLetters(int code) {
+        if (code - 10 <= 0) return "";
+        if (code - 10 == 1) return ", a";
+
+        return ", a-" + (char) (code - 9 + 'a');
+    }
+
+    public static int grader(Set<Character> correctAnswer, String input){
+        int bulls = 0;
+        int cows = 0;
+        int size = correctAnswer.size();
+        Iterator<Character> answerIterator = correctAnswer.iterator();
+
+        for (int i = 0; i < size; i++) {
+            if (answerIterator.next() == input.charAt(i))
+            {
+                bulls += 1;
+            } else if (correctAnswer.contains(input.charAt(i)))
+            {
+                cows += 1;
+            }
+        }
+
+        System.out.printf("Grade: %d bull(s) and %d cow(s).%n", bulls, cows);
+
+        return bulls;
+    }
+
+    public static boolean isUserAnswerCorrect(String userAnswer, int numberLength) {
+
+        // Check if user input exactly needed number of digits
+        if (userAnswer.length() != numberLength) {
+            System.out.printf("Please input exactly %d different digits.%n", numberLength);
+            return false;
+        }
+
+        // Check if all digits are different
+        for (char c : userAnswer.toCharArray())
+        {
+            if (userAnswer.length() - userAnswer.replaceAll(Character.toString(c),"").length() > 1)
+            {
+                System.out.println("Please ensure that all digits are different.");
+                return false;
+            }
+        }
+
+        // Check if there is no forbidden symbols
+
+        return true;
+    }
+
+    public static boolean wishToExit(String userAnswer) {
+        if (userAnswer.startsWith("/")) {
+            return "/exit".equals(userAnswer);
+        }
+
+        return false;
+    }
+
     public static void main(String[] args) {
-        System.out.print("Hello world!");
+
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Hello, player! Enter the length of the secret number:");
+        int secretNumberLength = scanner.nextInt();
+        scanner.nextLine();
+
+        System.out.println("Enter the number of unique symbols in code:");
+        int secretNumberOfSymbols = scanner.nextInt();
+        scanner.nextLine();
+
+        Set<Character> secretNumber = secretGenerator(secretNumberLength, secretNumberOfSymbols);
+
+        while (secretNumber.isEmpty())
+        {
+            System.out.println("Do you wish to input length once more? 1 - yes, other - no");
+            String wishAgain = scanner.nextLine().trim();
+
+            if ("1".equals(wishAgain))
```

Comment:
You may use a class from Java 8 called Objects which have a set of good method to compare and check objects